psycop/common/model_training_v2/hyperparameter_suggester/test_optuna_hyperparameter_search.py

```diff
@@ -0,0 +1,93 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import pytest
+from confection import Config
+from optuna import Trial
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.hyperparameter_suggester.optuna_hyperparameter_search import (
+    OptunaHyperParameterOptimization,
+)
+from psycop.common.model_training_v2.hyperparameter_suggester.suggesters.base_suggester import (
+    Suggester,
+)
+from psycop.common.model_training_v2.trainer.task.estimator_steps.logistic_regression import (
+    LogisticRegressionSuggester,
+)
+
+
+@BaselineRegistry.suggesters.register("mock_log_reg_suggester")
+@dataclass
+class MockLogisticRegression(Suggester):
+    c_high: float = 1
+
+    def suggest_hyperparameters(self, trial: Trial) -> dict[str, Any]:
+        return {
+            "@estimator_steps": "logistic_regression",
+            "C": trial.suggest_float("C", low=0.1, high=self.c_high, log=False),
+            "l1_ratio": 0.5,
+            "solver": "saga",
+            "penalty": "elasticnet",
+        }
+
+
+def test_validate_configspace_no_suggesters():
+    cfg = {"param1": 1, "param2": {"nested_param": 2}}
+
+    with pytest.raises(ValueError, match="No suggesters.*"):
+        OptunaHyperParameterOptimization()._validate_configspace(  # pyright: ignore[reportPrivateUsage]
+            cfg,
+        )
+
+
+def test_validate_configspace_with_suggester():
+    cfg = {
+        "param1": 1,
+        "param2": {
+            "nested_suggester": LogisticRegressionSuggester(
+                C={"low": 0.1, "high": 1, "logarithmic": False},
+                l1_ratio={"low": 0.1, "high": 1, "logarithmic": False},
+                solvers=("saga", "lbfgs"),
+            ),
+        },
+    }
+    assert (
+        OptunaHyperParameterOptimization()._validate_configspace(  # type: ignore
+            cfg,
+        )
+        is None
```

Comment:
Should this return anything? (Maybe we don't need the assertion, just a "not pytest.raises" or something?)

Reply:
Agreed, removed the assertion. The test will fail if an error is raised